indigo_app/management/commands/bulk_import_docx.py

```diff
@@ -0,0 +1,117 @@
+import csv, os
+from datetime import datetime
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from indigo.plugins import plugins
+from indigo_api.models import Document, Language, Work
+from indigo_api.serializers import AttachmentSerializer
+
+
+class Command(BaseCommand):
+    help = 'Imports new docx files on existing works at existing points in time. ' \
+           'Example: `python manage.py bulk_import_docx  ~/Namibia/namibia.csv ~/Namibia/STATUTES_DOCX`'
+
+    def add_arguments(self, parser):
+        parser.add_argument('csv_file', type=str,
+                            help='A path to a .csv file, e.g. `~/Namibia/namibia.csv`.'
+                                 'The file MUST have the following headings in the first row: '
+                                 '- frbr_uri (of existing work) '
+                                 '- date (format: 1985-10-17): this is the expression date of the doc '
+                                 '  and should be an existing PiT '
+                                 '- language: 3-letter code, e.g. `eng` for English '
+                                 '- filename: the name of the docx file to be imported from the directory.'
+                            )
+        parser.add_argument('path', type=str,
+                            help='a path to a directory that contains the docx files '
+                                 'given in the .csv file under `filename` '
+                            )
+
+    def get_user(self):
+        for user in User.objects.all().order_by('id'):
+            print('{}: {} {}'.format(user.id, user.first_name, user.last_name))
+        while True:
+            try:
+                result = int(input('Which user are you? Select the number from the list above: '))
+                user = User.objects.get(id=result)
+            except:
+                print('\nSomething went wrong; try again (you must type a number from the list above)\n\n')
+            else:
+                print('\nUser selected: {} {}.\n\n'.format(user.first_name, user.last_name))
+                return user
+
+    def get_file(self, i, path_to_filename):
+        try:
+            return open(path_to_filename)
+        except IOError as e:
+            print('\nERROR at row {}:'.format(i + 2))
+            print('File error: ' + str(e) + '\n')
+
+    def import_rows(self, user, csv_file, path):
+        content = csv.DictReader(csv_file)
+        for i, row in enumerate(content):
+            with transaction.atomic():
```

Comment:
This means that if row 1 succeeds but row 2 fails, row 1 is committed. Is that what you'd like? Another approach is all-or-nothing: wrap the entire batch in a transaction so if any one fails, everything is rolled back.

Reply:
it is what I had in mind but I'm open to being persuaded otherwise?

Reply:
I tend to prefer all-or-nothing. Let's go with what you have and change if necessary.